back-end/app/Http/Controllers/Admin/AdminController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Models\User;
+use Illuminate\Http\Request;
+use App\Http\Traits\AdminTrait;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Hash;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class AdminController extends Controller
+{
+    // Call traits for reusable codes for checking if the user is admin
+    use AdminTrait;
+
+    // Display a listing of the Users. 
+    public function index()
+    {
+        $users = User::where('is_admin', false)->get();
+
+        return response()->json(['data' => $users], 200);
+    }
+
+    // Display the specified User. 
+    public function show($user_id)
+    {
+        $user = User::findOrFail($user_id);
+
+        if ($user->is_admin) throw new HttpException(404, 'User not found');
+
+        return response()->json(['data' => $user], 200);
+    }
+
+    // Update the specified User in storage. 
+    public function update(Request $request, $user_id)
+    {
+        $this->isAdmin();
+
+        // Admin doesn't need to have a password confirmation function.
+
+        $user = $request->validate([
+            'first_name' => 'required|max:191',
+            'last_name' => 'required|max:191',
+            'email' => 'required|unique:users|max:191',
+            'password' => 'required',
+        ]);
+
+        User::findOrFail($user_id)->update([
+            'first_name' => $user['first_name'],
+            'last_name' => $user['last_name'],
+            'email' => $user['email'],
+            'password' => bcrypt($user['password']),
+        ]);
+
+        return response()->json(['message' => 'Account Updated Successfully'], 201);
+    }
```

Comment:
I don't think that we need this one. In real world, it is not practical that admin can change user's info.
But in change password, it is possible but it should generate random password.
Else, we can remove this feature.